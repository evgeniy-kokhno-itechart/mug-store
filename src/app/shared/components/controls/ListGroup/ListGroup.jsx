import React from 'react';
import { PropTypes } from 'prop-types';
import ListGroupItem from '../ListGroupItem/ListGroupItem';
import './ListGroup.scss';

const ListGroup = ({ items, selectedItem, onItemSelect }) => (
  <ul className="group-list">
    {items.map((item, index) => (
      <ListGroupItem
        key={item.id}
        item={item}
        index={index}
        selectedItem={selectedItem}
        handleItemSelect={onItemSelect}
      />
    ))}
  </ul>
);

ListGroup.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string, name: PropTypes.string })).isRequired,
  selectedItem: PropTypes.shape({ id: PropTypes.string, name: PropTypes.string }),
  onItemSelect: PropTypes.func.isRequired,
};

ListGroup.defaultProps = {
  selectedItem: null,
};

export default ListGroup;
