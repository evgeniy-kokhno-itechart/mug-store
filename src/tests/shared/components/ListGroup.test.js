/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ListGroup from '../../../app/shared/components/controls/ListGroup/ListGroup';

configure({ adapter: new Adapter() });

describe('<ListGroup />', () => {
  let ListGroupWrapper;
  const itemToBeselectedMock = { id: '2', name: 'test name 2' }; // this is to get reference equality in ListGroup component worked

  beforeEach(() => {
    const onItemSelect = jest.fn();
    ListGroupWrapper = shallow(<ListGroup items={[{ id: '1', name: 'test name 1' }, itemToBeselectedMock]} onItemSelect={onItemSelect} />);
  });

  it('renders with default props properly', () => {
    expect(ListGroupWrapper).toMatchSnapshot();
  });

  it('renders with second item selected', () => {
    ListGroupWrapper.setProps({ selectedItem: itemToBeselectedMock });
    expect(ListGroupWrapper).toMatchSnapshot();
  });
});
