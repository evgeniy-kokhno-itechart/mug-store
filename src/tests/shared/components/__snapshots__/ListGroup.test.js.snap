// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGroup /> renders with default props properly 1`] = `
<ul
  className="group-list"
>
  <ListGroupItem
    handleItemSelect={[MockFunction]}
    index={0}
    item={
      Object {
        "id": "1",
        "name": "test name 1",
      }
    }
    key="1"
    selectedItem={null}
  />
  <ListGroupItem
    handleItemSelect={[MockFunction]}
    index={1}
    item={
      Object {
        "id": "2",
        "name": "test name 2",
      }
    }
    key="2"
    selectedItem={null}
  />
</ul>
`;

exports[`<ListGroup /> renders with second item selected 1`] = `
<ul
  className="group-list"
>
  <ListGroupItem
    handleItemSelect={[MockFunction]}
    index={0}
    item={
      Object {
        "id": "1",
        "name": "test name 1",
      }
    }
    key="1"
    selectedItem={
      Object {
        "id": "2",
        "name": "test name 2",
      }
    }
  />
  <ListGroupItem
    handleItemSelect={[MockFunction]}
    index={1}
    item={
      Object {
        "id": "2",
        "name": "test name 2",
      }
    }
    key="2"
    selectedItem={
      Object {
        "id": "2",
        "name": "test name 2",
      }
    }
  />
</ul>
`;
